src/main/java/frc/trigon/robot/commands/VisualizeNoteShootingCommand.java

```diff
@@ -0,0 +1,94 @@
+        final Pose3d notePose = new Pose3d(fieldRelativeNoteExitPointTranslation, new Rotation3d()).plus(noteTransform);
+        noteZ = notePose.getTranslation().getZ();
+        Logger.recordOutput("Poses/GamePieces/ShotNotePose", notePose);
+    }
+
+    @Override
+    public boolean isFinished() {
+        return noteZ < 0;
+    }
+
+    @Override
+    public void end(boolean interrupted) {
+        Logger.recordOutput("Poses/GamePieces/ShotNotePose", new Pose3d(0, 0, 50, new Rotation3d()));
+    }
+
+    private Transform3d calculateNoteTransform(double timeDifference) {
+        return new Transform3d(calculateNoteXDifference(timeDifference), calculateNoteYDifference(timeDifference), calculateNoteZDifference(timeDifference), new Rotation3d(0, -startingPitch.getRadians(), 0));
+    }
+
+    private double calculateNoteXDifference(double t) {
+        return initialXYVelocity.getX() * t;
+    }
+
+    private double calculateNoteYDifference(double t) {
+        return initialXYVelocity.getY() * t;
+    }
+
+    private double calculateNoteZDifference(double t) {
+        return (initialZVelocity * t) + (-0.5 * ShootingConstants.G_FORCE * t * t);
+    }
+
+    private void configureStartingStats() {
+        startingTimeSeconds = Timer.getFPGATimestamp();
+
+        final Pose2d currentRobotPose = RobotContainer.POSE_ESTIMATOR.getCurrentPose();
+        final Rotation2d currentRobotAngle = currentRobotPose.getRotation();
+        final double startingTangentialVelocity = getStartingTangentialVelocity();
+
+        startingPitch = RobotContainer.PITCHER.getCurrentPitch();
+        fieldRelativeNoteExitPointTranslation = calculateFieldRelativeNoteExitPoint(currentRobotPose.getTranslation(), currentRobotAngle);
+        initialXYVelocity = calculateInitialXYVelocityWithRobotVelocity(startingPitch, startingTangentialVelocity, currentRobotAngle);
+        initialZVelocity = startingPitch.getSin() * startingTangentialVelocity;
+    }
+
+    private double getStartingTangentialVelocity() {
+        return SHOOTING_CALCULATIONS.angularVelocityToTangentialVelocity(RobotContainer.SHOOTER.getCurrentRightMotorVelocityRotationsPerSecond());
+    }
+
+    private Translation3d calculateFieldRelativeNoteExitPoint(Translation2d currentRobotPose, Rotation2d currentRobotAngle) {
+        final MirrorableRotation2d robotAngle = new MirrorableRotation2d(currentRobotAngle, false);
+        final Pose3d fieldRelativeNoteExitPoint = SHOOTING_CALCULATIONS.calculateShooterNoteExitPointFieldRelativePose(startingPitch, currentRobotPose, robotAngle);
+        return fieldRelativeNoteExitPoint.getTranslation();
+    }
+
+    private Translation2d calculateInitialXYVelocityWithRobotVelocity(Rotation2d startingPitch, double startingTangentialVelocity, Rotation2d currentRobotAngle) {
+        final Translation2d robotFieldRelativeTangentialVelocity = SHOOTING_CALCULATIONS.getRobotFieldRelativeVelocity().toTranslation2d();
```

Comment:
The robot's vel isn't tangential. Here you would need to write "meters per second", but as I said, unnecessary.